web/src/app/components/content/calculation/CalculationContainer.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import { EColor } from '../../../common/Color';
+import { IStandardTableRow, StandardTable } from '../../elements/Tables';
+import { H2, H3 } from '../../elements/Texts';
+import { IFluidInput, IMetricInput } from '../calculator/CalculateFormContainer';
+import { IShip } from '../calculator/ShipSection';
+import { IStandard } from '../calculator/StandardSection';
+import { ITransport } from '../calculator/TransportSection';
+
+export interface ICalculationContainer {
+    calculation: {
+        id: string,
+        fluid: IFluidInput;
+        ship: IShip;
+        standard: IStandard;
+        transport: ITransport;
+        result: IResult[];
+    };
+    style?: Object;
+}
+
+export const CalculationContainer = (props: ICalculationContainer) => {
+
+    var header = ["TIME", "WI", "GCV", "DENSITY", "TEMP", "VOLUME", "ENERGY"];
+    var rows: IStandardTableRow[] = [];
+    var initRow: IResult;
+    var endRow: IResult;
+    var cardResults: ICardResult[] = [];
+
+    //table
+    props.calculation.result.forEach((result: IResult, index: number) => {
+        if (index === 0) {
+            initRow = result;
+        }
+        else if (index === props.calculation.result.length - 1) {
+            endRow = result;
+        }
+        var row: IStandardTableRow = {
+            value: result.id,
+            display: [
+                result.time.value,
+                result.wi.value,
+                result.gcv.value,
+                result.density.value,
+                result.temp.value,
+                result.volume.value,
+                result.energy.value
+            ]
+        }
+        rows.push(row);
+    })
+
+    //card result (TODO: improve)
+    header.forEach((title:string, index) =>{
+        cardResults.push({
+            title: title,
+            unit: initRow[title.toLowerCase() as never]["unit"],
+            initValue: initRow[title.toLowerCase() as never]["value"],
+            endValue: endRow[title.toLowerCase() as never]["value"]
+        })
+    })
```

Comment:
Could use a map instead

```javascript
const cardResult: ICardResult = headers.map(title => ({
      title,
      unit: initRow[title.toLowerCase()]["unit"],
      initValue: initRow[title.toLowerCase()]["value"],
      endValue: endRow[title.toLowerCase()]["value"]
}));
```

Is `never` correct here? If a type is needed, why not string?